[WIP] Incorporating product terms into a single A-Z term list

bergerhoffer opened the pull request:
Preview: http://file.rdu.redhat.com/~ahoffer/2022/ssg_alphabetize/master.html

A few notes:
* To pilot this, I've taken the first two terms of each product-specific file and have moved them to the appropriate general letter file.
* Any terms that are product-specific, I added wording like "In \<product>, ..." to the front of the description. I noted what I added in a comment above the entry to make it easier to review, as opposed to reviewing the entire entry.
* Please ignore any changes in the product-specific files themselves. Terms that I've moved are commented out for tracking purposes at the moment
* Note that some "See also" links won't work right now because not all terms have been included yet.

We are moving forward with this change. Team is helping with evals, then once those are done I'll update this PR with the changes. Tracking which products have been updated here:

- [x] AMQ
- [x] Azure
- [x] BxMS
- [x] Ceph
- [x] CloudForms
- [x] Data Grid
- [x] Fuse
- [x] JBoss EAP
- [x] OCS
- [x] OpenShift
- [x] OpenStack
- [x] RHDS
- [x] RHEL
- [x] Satellite
- [x] SSO
- [x] RHV

This PR is just for reorganizing the terms. Any followup issues found during the reorg are being captured in #180 and will be fixed later.

leswilliams44 commented:
Thanks for doing this, @bergerhoffer. It should make it easier to find terms.

dfitzmau commented:
Thanks, @bergerhoffer . Looks very good, and is much easier when trying to locate a term. Thank you!

mportman12 commented:
@bergerhoffer Looks good! My only concern is that it's easy to overlook the "In [product]" intro and miss that the definition or term is product-specific. Did you play-test adding a Product field for the relevant terms?

bergerhoffer commented:
> @bergerhoffer Looks good! My only concern is that it's easy to overlook the "In [product]" intro and miss that the definition or term is product-specific. Did you play-test adding a Product field for the relevant terms?

@mportman12 I had actually played with this when I was trying out adding an update to make the product-specific entries stand out more (before we decided to try the single A-Z list). See [this PR](https://github.com/redhat-documentation/supplementary-style-guide/pull/162) for some examples. Outside of those examples, I did try to add a new "Product:" field, but I really didn't like how it looked, so I didn't include it as an option :smile: 

If we're moving to a single A-Z list, I think that it's somewhat less important to have it stand out so much that the term applies to a particular product. I think it's normal for terms in a glossary to be specific to a certain situation.

For example, here are two terms in our generic list now that have a "In \<x>," to begin.

![image](https://user-images.githubusercontent.com/7443887/164527033-6d53fafc-62e4-460c-8573-b71f8ac0f88b.png)

![image](https://user-images.githubusercontent.com/7443887/164527102-f5683ae7-7561-48a0-b93e-f96ed909da54.png)

Ours would be the same kind of thing, but just start with a specific product, if applicable.

mportman12 commented:
@bergerhoffer Ok, then no further objections!

bredamc commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/a.adoc:
```diff
@@ -60,6 +170,30 @@ The AMD64 logo is trademarked; the term "AMD64" is not trademarked. For more inf
+[discrete]
+[[jboss-amq]]
+==== image:images/yes.png[yes] AMQ (noun)
+*Description*: The short product name for Red Hat AMQ.
```
The description refers to "Red Hat AMQ", but that term is also listed in the "Incorrect forms" section. Should one of these sections be updated?

bergerhoffer replied:
That is really odd, definitely seems wrong to me. I'm hesitant to make any changes outside of reorganizing the entries. 

I'll have to see what all comes up - maybe we can capture these types of things into an issue that people add to when they come across things, and we do a sweep after the reorg.

mportman12 replied:
@bredamc Based on the text of the Red Hat AMQ entry, I think it was an error to list "AMQ" in "Incorrect forms". The last time AMQ came up in Style Council, @bergerhoffer used someone named Fintan as a contact. Andrea, can Fintan assist again on this?

bredamc commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/a.adoc:
```diff
@@ -71,6 +205,18 @@ For more information about Intel® trademarks, see the http://www.intel.com/cont
 
 *See also*:
 
+// CloudForms: Added "In Red Hat CloutForms"
```
```suggestion
// CloudForms: Added "In Red Hat CloudForms"
```

bredamc commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/a.adoc:
```diff
@@ -9,6 +9,116 @@
+[discrete]
+[[action]]
+==== image:images/yes.png[yes] action (noun)
+*Description*: In the context of authorization in OpenShift Container Platform, an action consists of _project_, _verb_, and _resource_.
```
```suggestion
*Description*: In OpenShift Container Platform, an action consists of _project_, _verb_, and _resource_.
```
In the interests of minimalism, is it ok to delete the extra text?

bergerhoffer replied:
I don't think I would remove it, because noting that it's related to authorization is probably important. But I do think that we could move it elsewhere into the sentence so that "In OpenShift Container Platform" is first.

rolfedh replied:
```suggestion
*Description*: In OpenShift Container Platform, an authorization action consists of _project_, _verb_, and _resource_.
```

bredamc reviewed (commented):
@bergerhoffer Apologies for the delay. I think your approach is fantastic, very well organised. I especially liked how you prefixed each change with a comment, thank you!

I think for consistency, it would be best to prefix all product-specific entries with "In <product>, ...", which might mean rephrasing some of the existing entries that refer to the product name later in the sentence.

I started to add some other review comments, but I'm not sure that's what you want at this stage, so I'll wait for now. Great job again :)

bergerhoffer commented:
> @bergerhoffer Apologies for the delay. I think your approach is fantastic, very well organised. I especially liked how you prefixed each change with a comment, thank you!

@bredamc Yeah I did this mostly for review purposes (so that people aren't reviewing the entire text that looks new, but was just moved). I'd like to minimize the changes in this PR to not making major wording changes (outside of the "In \<product>" rewords). I was debating on whether to leave those comments in there, or remove them after reviews but before merging. Undecided on that right now :smile: 

> I think for consistency, it would be best to prefix all product-specific entries with "In , ...", which might mean rephrasing some of the existing entries that refer to the product name later in the sentence.

@bredamc Thanks for the input! I hadn't rephrased anything that already mentioned the product, but the longer I've been thinking on it, I do agree that it would be best to rephrase so that the "In \<product>," is at the very front of each description.

themr0c commented:
Have you considered having one reference module file per word, and the index per alpha forming assembly? I believe it would be a good choice for maintainability. (Additionally the index per alpha could be generated)

inoxx03 commented:
> @bergerhoffer Looks good! My only concern is that it's easy to overlook the "In [product]" intro and miss that the definition or term is product-specific. Did you play-test adding a Product field for the relevant terms?

I think you make a good point @mportman12, I've been thinking the same thing.
Having the product designation incorporated in the text of the term definition makes for good readability, but on the other hand, I think having a _Product_ field on each entry would make the entries easier to scan.

I think adding a label or "badge" with the product name to the title of an entry could achieve the same.

Some entries might have multiple definitions, some of which might general, while others might be product-specific. We could label product-specific definitions with the name of the products that they apply to.

As a side note, I call them "labels" but, we could just use regular formatted product names.
Possibly, store the names in standard AsciiDoc attributes.

For example:
```
// bold formatting is optional
:label-product-rhamq: **Red Hat AMQ**
```

@bergerhoffer @mportman12 WDYT

bergerhoffer commented:
> Have you considered having one reference module file per word, and the index per alpha forming assembly? I believe it would be a good choice for maintainability. (Additionally: the index per alpha could be generated, and it would open the possibility to write tests validating the structural consistency for the per-word-reference-modules).

@themr0c I know I responded to you on this in that Slack thread, but posting here as well for tracking. No we haven't considered that, but IMO having a separate file per word seems like much more overhead than necessary, as opposed to breaking it down by letter.

bergerhoffer commented:
Pushed updates for the first set of products: Azure, CloudForms, Data Grid, and Satellite. Will add the checklist of products to the main description to track status.

bergerhoffer commented:
Initial reorganization for all products is complete. This is ready for review.

Preview as of today is here: http://file.rdu.redhat.com/~ahoffer/2022/master.html

mportman12 commented:
In the process of checking alphabetizing. In the meantime, I noticed the following issues:

- [ ] The following terms contain a "Class" metadata that should be removed:
- libvirt
- MDS
- RADOS

- [ ] The terms are missing parts of speech in the term header:
- access token
- assertion
- authentication
- authentication flow
- client
- client adapter
- client role
- client scope
- composite role
- consent
- credentials
- direct grant
- group
- identity provider
- identity provider federation
- identity provider mappers
- identity token
- network interface card
- network interface controller (NIC)
- NIC
- protocol mapper
- realm
- Red Hat Enterprise Linux
- required action
- role
- service account
- session
- SmartNIC
- theme
- user federation provider
- user role mapping

bergerhoffer commented:
@mportman12 thanks a lot for pointing these out!

I addressed the first and third items (removing the class field and fixing the alphabetical orders).

I had noticed the second item as well, and it's captured in #180 (third checkbox) to address after this reorg. It was all RHSSO terms that were missing the part of speech. Also, in case you notice it later as well, all of OCS terms are missing the appropriate icon.

Can you rerun your report on the alphabetical comparison and make sure that it all looks good now? I also updated the preview file: http://file.rdu.redhat.com/~ahoffer/2022/master.html

Also, some of these were ones that were out of order prior to my reorg. I'm thinking that it might be good to add some guidance on checking alphabetical order when adding new entries. Maybe somewhere in the guidelines on adding new entries, you could make an update to what alphabetizer check you used, so that we can make sure that things going forward are in the correct order. wdyt?

bredamc commented:
The ISG advice re. sorting is the same as that used by Google Docs:
"A two-word version should be listed before a hyphenated version, which in turn should be listed before a one-word version.
Correct example: off line, off-line, offline"

jherrman commented:
Hi Andrea, definitely a good change for the overall readability of the guide.

One (very cosmetic) modification I'd propose is changing the part-of-speech classifications such as `(noun)` to italics. This will distinguish that bit of the "bridgehead" more visibly from the actual term, and thus make it more easily readable.

ndeevy commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/b.adoc:
```diff
@@ -209,6 +305,54 @@ The practice of having both modes together is often referred to as "hybrid", "ag
+[discrete]
+[[broker-cluster]]
+==== image:images/yes.png[yes] broker cluster (noun)
+*Description*: A group of brokers to be grouped together in order to share message processing load. In JBoss A-MQ 6, this was called a _network of brokers_.
```
Should this start as a full sentence? And do we need the 'in order to' or the 'to be' future/passive bit?:
Maybe this is overkill or incorrect but suggesting:
`A broker cluster is a group of brokers that can share message processing load.`

ndeevy commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/b.adoc:
```diff
@@ -241,3 +445,63 @@ The practice of having both modes together is often referred to as "hybrid", "ag
+[discrete]
+[[business-process]]
+==== image:images/yes.png[yes] business process (noun)
+*Description*: A "business process" is a collection of related, structured tasks that results in achieving a specific target. It is presented as as a flowchart comprising a sequence steps necessary to achieve business goals.
```
```suggestion
*Description*: A "business process" is a collection of related, structured tasks that results in achieving a specific target. It is presented as a flowchart comprising a sequence of steps necessary to achieve business goals.
```

IngridT1 replied:
I would avoid the word "comprising". You have used it correctly, but most people do not. How about:

"It is presented as a flowchart that consists of a sequence of steps"

ndeevy commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/c.adoc:
```diff
@@ -65,6 +113,185 @@
+[discrete]
+[[certificate-authorities]]
+==== image:images/yes.png[yes] certificate authorities (noun)
+*Description*: An entity that issues digital certificates. In Red Hat Identity Management, the primary CA is `ipa`, the IdM CA. The `ipa` CA certificate is one of the following types:
```
Maybe the entry should be singular:
`A certificate authority (CA), is an entity that issues..`

For sentence completeness and to define the abbreviation

lburcher7 commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/o.adoc:
```diff
@@ -1,3 +1,75 @@
+[discrete]
+[[object-store]]
+==== image:images/yes.png[yes] Object Store (noun)
+*Description*: In Red Hat Ceph Storage, Object Store is a core component of the Ceph Storage Cluster. Also referred as RADOS.
```
Typo: it should be "referred to as". This is not a full sentence either, so maybe add "It's" or "This component is" before "also referred to as RADOS".

ndeevy commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/c.adoc:
```diff
@@ -65,6 +113,185 @@
+* Self-signed. In this case, the `ipa` CA is the root CA.
+* Externally signed. In this case, the `ipa` CA is subordinated to the external CA.
+--
+In IdM, you can also create multiple *sub-CAs*. Sub-CAs are IdM CAs whose certificates are one of the following types:
```
I don't think 'sub-CAs' should be in bold font.

lburcher7 commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/o.adoc:
```diff
@@ -131,14 +215,89 @@
+- OpenShift Dedicated
+- OpenShift Kubernetes Engine
+
+Previously, the upstream distribution was called OpenShift Origin, however it is now called OKD; use of the OpenShift Origin name is deprecated.
```
Colon needed before "however". Here's a suggestion to simplify the sentence overall.

The [current] name of the upstream distribution is OKD.  The name OpenShift Origin is deprecated.

lburcher7 commented on supplementary_style_guide/glossary_terms_conventions/general_conventions/p.adoc:
```diff
@@ -64,6 +140,30 @@
+[discrete]
+[[picketlink-federation]]
+==== image:images/yes.png[yes] picketlink-federation subsystem (noun)
+*Description*: In Red Hat JBoss Enterprise Application Platform, the "picketlink-federation" subsystem is used to configure single sign-on (SSO) using security assertion markup language (SAML). In general text, write in lowercase as two words separated by a hyphen. Use "PicketLink Federation subsystem" when referring to the picketlink-federation subsystem in titles and headings. When writing the  ...
```
Should "security assertion markup language" be capitalized? According to the IBMSG, "Most spelled-out forms of language and protocol names have initial uppercase letters." Also, I do have the same question as with TOML about whether we should spell it out at all.

IngridT1 commented:
For the overall glossary, can we list two different definitions for the same term as an ordered list instead of two sentences embedded in a paragraph? That would be easier to read. 

Also, @lburcher7 and I think we should tag the definitions with multiple meanings in some way. For example, you could have the term "bucket" and after Description: include (multiple definitions) in parentheses. 

**EXAMPLE**

bucket (noun)

Description (multiple definitions):
1) In the S3 API, a bucket contains objects. A bucket also defines access control lists (ACLs). Unlike folders or directories, buckets cannot contain other buckets. A bucket in the S3 API is synonymous with a "container" in the Swift API. 

2) In a CRUSH hierarchy, the term bucket is sometimes used. But CRUSH buckets and S3 buckets are mutually exclusive concepts.

Use it: yes

Incorrect forms:

See also: [container](http://file.rdu.redhat.com/~ahoffer/2022/master.html#container)

bergerhoffer commented:
@jherrman Thanks for the suggestion, I'll bring that up with the team on whether we should italicize the part of speech.

@lburcher7 @IngridT1 @ndeevy Thanks for the wording suggestions, however these were just straight copy/paste from other files. I specifically avoided changing any of the wording when possible, to keep this just a clean reorganization. We can absolutely address your suggestions - but we'll do them in a separate PR afterward. During the reorg, I captured other items to follow up on afterward in #180. I'll make sure we also come back to look at these wording suggestions.

@IngridT1 your thoughts on the entries with multiple descriptions is definitely something we can investigate too. I did actually add something similar to follow up on afterward in #180, since there were varying formats. So I'll update that item to look at your suggestions when considering which format to use.

bergerhoffer commented:
Since the only feedback left are wording suggestions and other items that we'll follow up with separately, is this PR good enough to be merged now? CC @redhat-documentation/ccs-style-council

joaedwar reviewed (approved):
LGTM

bergerhoffer commented:
Merging since there are 3 approvals, and so that we can get started working on all of the followup items.
